dashboard/run_old_data_collection.py

```diff
@@ -0,0 +1,130 @@
+    ) - timedelta(days=day_shift)
+    utc_start_datetime = local_start_datetime - DataCollector.utc_time_gap
+    start_utc_time_string = utc_start_datetime.strftime(DataCollector.utc_time_format)
+    end_utc_time_string = (utc_start_datetime + timedelta(days=1)).strftime(
+        DataCollector.utc_time_format
+    )
+    return local_start_datetime.date(), start_utc_time_string, end_utc_time_string
+
+
+def collect_old_data(
+    dashboard_db: DashboardDB,
+    hpc: const.HPC,
+    login_user: str,
+    users: Iterable[str],
+    day_shift: int,
+):
+    """Collect old core hours usage for a specified users in a previous day"""
+    local_date, start_time, end_time = get_old_utc_times(day_shift)
+    cmd = "sreport -M {} -t Hours cluster AccountUtilizationByUser Users={} start={} end={} -n".format(
+        hpc.value, " ".join(users), start_time, end_time
+    )
+    user_ch_output = (
+        subprocess.check_output(
+            "ssh {}@{} {}".format(login_user, hpc.value, cmd), shell=True, timeout=10
+        )
+        .decode("utf-8")
+        .strip()
+        .split("\n")
+    )
+    if user_ch_output:
+        dashboard_db.update_user_chours(
+            hpc,
+            DataCollector.parse_user_chours_usage(user_ch_output, users, local_date),
+            local_date,
+        )
+
+
+def run_old_collection(dashboard_db, hpcs: Union[List[const.HPC], const.HPC], login_user: str, users: Iterable[str], days_shift: int):
+    """Runs the data collection for a specified period"""
+    # Iterate through the specifid days period
+    hpcs = [hpcs] if type(hpcs) is const.HPC else hpcs
+    for day_shift in range(days_shift + 1):
+        # Collect the data
+        print("{} - Collecting data from HPC {}".format(datetime.now(), hpcs))
+        for hpc in hpcs:
+            collect_old_data(dashboard_db, hpc, login_user, users, day_shift)
+            print("{} - Done".format(datetime.now()))
+
+
+def main(args):
+    hpc = (
+        const.HPC(args.hpc)
+        if type(args.hpc) is str
+        else [const.HPC(cur_hpc) for cur_hpc in args.hpc]
+    )
+    if not os.path.isfile(args.dashboard_db):
+        dashboard_db = DashboardDB.create_db(args.dashboard_db)
+    else:
+        dashboard_db = DashboardDB(args.dashboard_db)
+
+    run_old_collection(dashboard_db, hpc, args.user, args.users, args.days_shift)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "user",
+        help="Username to run the data collection under, needs to have an open ssh socket to the HPC(s)",
+    )
+    parser.add_argument("dashboard_db", help="The dashboard db to use")
+    parser.add_argument(
+        "--update_interval",
+        help="Interval between data collection (seconds)",
+        default=30,
+    )
+    parser.add_argument(
+        "--hpc",
+        choices=HPCS,
+        help="Specify the HPCs on which to collect data, defaults to all HPCs",
```

Comment:
Should have nargs="+" set to allow specifying multiple, or reword to be singular.

Reply:
--hpc is default to mutilple, if you decide to specify -hpc, only one hpc should be specified